src/app/embed/adapters/qsd.adapter.ts

```diff
@@ -0,0 +1,58 @@
+import { EMBED_AUDIO_URL_PARAM, EMBED_TITLE_PARAM, EMBED_SUBTITLE_PARAM,
+  EMBED_SUBSCRIBE_URL_PARAM, EMBED_SUBSCRIBE_TARGET, EMBED_IMAGE_URL_PARAM } from './../embed.constants';
+import { Observable } from 'rxjs/Observable';
+
+export class QSDAdapter {
+  constructor(private params: Object) {
+    this.params = params
+  }
+
+  get hasFullInformation(): boolean {
```

Comment:
I don't think this method is useful if you're thinking about this problem correctly.